### ISSUE TITLE
[Breaking change]: FileSystemEventArgs.FullPath returns a truly fully qualified path

### ISSUE THREAD

**Issue by slask** [dotnet/docs#27916]
### Description

Initially the `FullPath` property of `FileSystemEventArgs` and `RenamedEventArgs` returned whatever directory path it was passed in from `FileSystemWatcher`. That could mean even a relative path, that does not match with the true intent and name of the **FullPath** and **OldFullPath** properties on the event args classes.

### Version

.NET 7 Preview 1
Reverted in .NET 7 Preview 5

### Previous behavior

`FullPath` and `OldFullPath` simply mirror what is was passed into FileSystemWatcher initially, which can include a relative path.

### New behavior

The `FullPath` and `OldFullPath` will now return a fully qualified path even when FileSystemWatcher is initialized with a relative path.

### Type of breaking change

- [ ] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load/execute or different run-time behavior.
- [X] **Source incompatible**: Source code may encounter a breaking change in behavior when targeting the new runtime/component/SDK, such as compile errors or different run-time behavior.

### Reason for change

To align the actual result with the intent and name of the `FullPath`/`OldFullPath` properties from the event args.

### Recommended action

The users will have to check their usage of FullPath and OldFullPath when dealing with FileSystemEventArgs and RenamedEventArgs objects and make sure they account for the path change.

### Feature area

Core .NET libraries

### Affected APIs

Only the FullPath and OldFullPath properties are affected.

**Comment by slask** [comment]
The PR with the fix: https://github.com/dotnet/runtime/pull/63051

**Comment by jeffhandley** [comment]
We decided to revert this breaking change, and the change back will be included in .NET 7 Preview 5. Here's the context for that decision: https://github.com/dotnet/runtime/issues/62606#issuecomment-1118113636

@gewarren, I will submit a PR to add additional API docs around the `FullPath` and `OldFullPath` properties to explain their behavior more accurately. Should the breaking change doc hat was added be deleted or should it be retained and updated to state the change was reverted?
